meshtastic/util.py

```diff
@@ -545,3 +551,19 @@ def detect_windows_port(sd):
                 #print(f'x:{x}')
                 ports.add(f'COM{x}')
     return ports
+
+
+def check_if_newer_version():
+    """Check pip to see if we are running the latest version."""
```

Comment:
Please introduce a cache for the check result. Using this on a computer without active net connection will stall with every CLI call till the connect timeout is reached.

Reply:
I did measurements under Linux and it is faster without an internet connection (0.341s vs. 0.254s):

```
$ time python -m meshtastic --support

If having issues with meshtastic cli or python library
or wish to make feature requests, visit:
https://github.com/meshtastic/python/issues
When adding an issue, be sure to include the following info:
 System: Linux
   Platform: Linux-6.2.7-200.fc37.x86_64-x86_64-with-glibc2.36
   Release: 6.2.7-200.fc37.x86_64
   Machine: x86_64
   Encoding (stdin): utf-8
   Encoding (stdout): utf-8
 meshtastic: v2.1.3 (*** newer version v2.1.4 available ***)
 Executable: /home/manuel/Documents/PlatformIO/Projects/python-fork/meshtastic/__main__.py
 Python: 3.11.2 CPython GCC 12.2.1 20221121 (Red Hat 12.2.1-4)

Please add the output from the command: meshtastic --info

real    0m0.341s
user    0m0.238s
sys     0m0.056s

$ time python -m meshtastic --support

If having issues with meshtastic cli or python library
or wish to make feature requests, visit:
https://github.com/meshtastic/python/issues
When adding an issue, be sure to include the following info:
 System: Linux
   Platform: Linux-6.2.7-200.fc37.x86_64-x86_64-with-glibc2.36
   Release: 6.2.7-200.fc37.x86_64
   Machine: x86_64
   Encoding (stdin): utf-8
   Encoding (stdout): utf-8
 meshtastic: v2.1.3
 Executable: /home/manuel/Documents/PlatformIO/Projects/python-fork/meshtastic/__main__.py
 Python: 3.11.2 CPython GCC 12.2.1 20221121 (Red Hat 12.2.1-4)

Please add the output from the command: meshtastic --info

real    0m0.254s
user    0m0.205s
sys     0m0.048s
$

```

Reply:
I did measurements under Linux and it is faster without an internet connection (0.341s vs. 0.254s). It does affect only meshtastic --info and --support command, but not any other.

```
$ time python -m meshtastic --support    # test with connection

If having issues with meshtastic cli or python library
or wish to make feature requests, visit:
https://github.com/meshtastic/python/issues
When adding an issue, be sure to include the following info:
 System: Linux
   Platform: Linux-6.2.7-200.fc37.x86_64-x86_64-with-glibc2.36
   Release: 6.2.7-200.fc37.x86_64
   Machine: x86_64
   Encoding (stdin): utf-8
   Encoding (stdout): utf-8
 meshtastic: v2.1.3 (*** newer version v2.1.4 available ***)
 Executable: /home/manuel/Documents/PlatformIO/Projects/python-fork/meshtastic/__main__.py
 Python: 3.11.2 CPython GCC 12.2.1 20221121 (Red Hat 12.2.1-4)

Please add the output from the command: meshtastic --info

real    0m0.341s
user    0m0.238s
sys     0m0.056s

$ time python -m meshtastic --support    # test without connection

If having issues with meshtastic cli or python library
or wish to make feature requests, visit:
https://github.com/meshtastic/python/issues
When adding an issue, be sure to include the following info:
 System: Linux
   Platform: Linux-6.2.7-200.fc37.x86_64-x86_64-with-glibc2.36
   Release: 6.2.7-200.fc37.x86_64
   Machine: x86_64
   Encoding (stdin): utf-8
   Encoding (stdout): utf-8
 meshtastic: v2.1.3
 Executable: /home/manuel/Documents/PlatformIO/Projects/python-fork/meshtastic/__main__.py
 Python: 3.11.2 CPython GCC 12.2.1 20221121 (Red Hat 12.2.1-4)

Please add the output from the command: meshtastic --info

real    0m0.254s
user    0m0.205s
sys     0m0.048s
$

```

Reply:
If it's failing that fast it's a non issue indeed... :-)